Support shebang parsing

kunli2 opened the pull request:
fixes https://github.com/openrewrite/rewrite-kotlin/issues/569

Following the same pattern in groovy, added a nullable string to `K.CompilationUnit` for shebang.

knutwannheden reviewed (approved):
Can we add more tests?

kunli2 commented:
> Can we add more tests?

Sure, added more tests, and the scenarios that shebang following with package declaration or imports need to be tackled a bit more, updated.
